File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Ước chung lớn nhất 2/UCLN2.cpp
#include<iostream>
using namespace std;
int UCLN(long long a, long long b)
{
    if(b == 0) 
        return a;
    return UCLN(b, a % b);
}
int main()
{
    long long a, b;
    cin >> a >> b;
    cout << UCLN(a, b);
    return 0;
}

File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Ước chung lớn nhất 2/UCLN2.cs
using System;

namespace UCLN
{
    public class Program
    {
        public static long UCLN(long  a, long  b)
        {
            if (b == 0)
                return a;
            return UCLN(b, a % b);
        }

        public static void Main(string[] args)
        {
            string []input = Console.ReadLine().Split(' ');
            long a = long.Parse(input[0]);
            long b = long.Parse(input[1]);
            Console.Write(UCLN(a, b));
        }
    }
}
